community/jmx/src/main/java/org/neo4j/jmx/impl/KernelBean.java

```diff
@@ -54,6 +54,25 @@
         this.instanceId = kernel.instanceId();
         this.query = support.createMBeanQuery( instanceId );
 
+        // perhaps not the most elegant way to tell which edition we are running...
+        String edition;
+        switch ( kernel.graphDatabase().getClass().getSimpleName() )
```

Comment:
Is it possible to add `GraphDatabaseAPI#getEdition()` and avoid this class name check?